edgepi-thermocouple.js

```diff
@@ -1,43 +1,52 @@
-module.exports = function(RED) {
-    const rpc = require("@edgepi-cloud/edgepi-rpc")
+module.exports = function (RED) {
+  const rpc = require("@edgepi-cloud/edgepi-rpc");
 
-    function ThermocoupleNode(config) {
-        // Create node with user configs
-        RED.nodes.createNode(this, config);
-        const node = this;
-        const ipc_transport = "ipc:///tmp/edgepi.pipe"
-        const tcp_transport = `tcp://${config.tcpAddress}:${config.tcpPort}`
-        const transport = (config.transport === "Network") ? tcp_transport : ipc_transport;
+  function ThermocoupleNode(config) {
+    RED.nodes.createNode(this, config);
+    const node = this;
+    let output = config.output;
 
-        // Init new tc
-        const tc = new rpc.TcService(transport)
-        if (tc){
-            console.debug("Thermocouple node initialized on: ", transport);
-            node.status({fill:"green", shape:"ring", text:"tc initialized"});
+    initializeNode(config).then((tc) => {
+      this.on("input", async function (msg, send, done) {
+        node.status({ fill: "green", shape: "dot", text: "input recieved" });
+        try {
+          output = msg.payload ?? output;
```

Comment:
Question : what is the possible value of `msg.payload`?
In the Nullish Operator (??) here: This operator returns the right value (`output`) ONLY if the left value is either `null` or `undefined`.
So if the `msg.payload` is neither `null` nor `undefined` (empty string , 0 ) this operator won't return the right value (`output`).
I would recommend to use `||` in this case